agent/consul/server.go

```diff
@@ -1552,8 +1553,29 @@ func (s *Server) RPC(ctx context.Context, method string, args interface{}, reply
 		metrics.IncrCounter([]string{"client", "rpc", "exceeded"}, 1)
 		return structs.ErrRPCRateExceeded
 	}
+
+TRY:
```

Comment:
Thinking more about this, retrying here does not make any sense. As the error is retry elsewhere and we can't target another server from this server (local request). The error should be returned to the caller to retry elsewhere. sorry for the confusion again! 😓